src/App.jsx

```diff
@@ -14,45 +17,124 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+
+function getNewSort(goods, alphOrLength, reversed) {
+  const newGoods = [...goods];
+
+  if (!alphOrLength && reversed) {
+    return newGoods.reverse();
+  }
+
+  if (alphOrLength) {
+    newGoods.sort((good1, good2) => {
+      switch (alphOrLength) {
+        case SORT_FIELD_LENGTH:
+          if (good1.length !== good2.length) {
+            return reversed
+              ? good2.length - good1.length
+              : good1.length - good2.length;
+          }
+
+          return reversed
+            ? good2.localeCompare(good1)
+            : good1.localeCompare(good2);
+
+        case SORT_FIELD_ALPHABET:
+          return reversed
+            ? good2.localeCompare(good1)
+            : good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return newGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [sortReverse, setSortReverse] = useState(false);
+  const updateGoods = getNewSort(goodsFromServer, sortField, sortReverse);
+
+  const newSortField = (field) => {
+    setSortField(field);
```

Comment:
The 'newSortField' function should be wrapped in a useCallback hook to avoid unnecessary re-renders as the function will not be recreated every time the component re-renders.